src/services/modules.js

```diff
@@ -0,0 +1,170 @@
+// Dependencies
+const httpStatusCode = require('@generics/http-status')
+const common = require('@constants/common')
+const modulesQueries = require('@database/queries/modules')
+const permissionsQueries = require('@database/queries/permissions')
+const { UniqueConstraintError, ForeignKeyConstraintError } = require('sequelize')
+const { Op } = require('sequelize')
+
+module.exports = class modulesHelper {
+	/**
+	 * Create modules.
+	 * @method
+	 * @name create
+	 * @param {Object} bodyData - modules body data.
+	 * @param {String} id -  id.
+	 * @returns {JSON} - modules created response.
+	 */
+
+	static async create(bodyData) {
+		try {
+			const modules = await modulesQueries.createModules(bodyData)
+			return common.successResponse({
+				statusCode: httpStatusCode.created,
+				message: 'MODULES_CREATED_SUCCESSFULLY',
+				result: {
+					Id: modules.id,
+					code: modules.code,
+					status: modules.status,
+				},
+			})
+		} catch (error) {
+			if (error instanceof UniqueConstraintError) {
+				return common.failureResponse({
+					message: 'MODULES_ALREADY_EXISTS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+			throw error
+		}
+	}
+
+	/**
+	 * Update modules.
+	 * @method
+	 * @name update
+	 * @param {Object} bodyData - modules body data.
+	 * @param {String} _id - modules id.
+	 * @param {String} loggedInUserId - logged in user id.
+	 * @returns {JSON} - modules updated response.
+	 */
+
+	static async update(id, bodyData) {
+		try {
+			const modules = await modulesQueries.findModulesById(id)
+			if (!modules) {
+				throw new Error('MODULES_NOT_FOUND')
+			}
+			const filter = { id }
+			const oldModuleCode = { module: modules.code }
+			const updatedModules = await modulesQueries.updateModules(filter, bodyData)
+			const newModuleCode = { module: updatedModules.code }
```

Comment:
@sumanvpacewisdom avoid declaring the new variables. can be achieved without that also